src/operator/nn/upsampling-inl.h

```diff
@@ -84,6 +86,21 @@ struct UpSamplingParam : public dmlc::Parameter<UpSamplingParam> {
   }
 };  // struct UpSamplingParam
 
+inline std::vector<int> scaleComp(const UpSamplingParam &param) {
+  std::vector<int> scaleArr{ 1, 1 };
+  if (param.scale.ndim() == 1) {
+    scaleArr[0] = param.scale[0];
+    scaleArr[1] = param.scale[0];
+  } else if (param.scale.ndim() == 2) {
+    scaleArr[0] = param.scale[0];
+    scaleArr[1] = param.scale[1];
+  } else if (param.scale.ndim() == 4) {
+    scaleArr[0] = param.scale[2];
+    scaleArr[1] = param.scale[3];
+  }
```

Comment:
What happens when scale is a single scalar rather than a Tuple/TShape?

Reply:
I believe when a single scalar is passed, it is a tuple of ndim == 1 with the scalar value in param.scale[0]

Reply:
Thanks for the offline debug, we found that passing in a scalar results in a TShape with ndim=1 and the value in index 0. So at least its working